### PR TITLE
DEPR: groupby.fillna

### LINKED ISSUE #48395: DEPR: GroupBy.fillna positional arguments
All positional arguments to Series.fillna are deprecated except for value and must be keyword only in the future. The corresponding change should be made to SeriesGroupBy and DataFrameGroupBy implementations of fillna (and perhaps on other objects as well).

### LINKED ISSUE #48792: DEPR: inplace in DataFrameGroupBy.fillna and SeriesGroupBy.fillna
Specifying `inplace` returns an empty frame or a Series with all `None` values, and neither modifies the original input. Instead of fixing this, I think we should deprecate the `inplace` argument of these methods.

```
df = pd.DataFrame({'a': [1, 1, 2], 'b': [np.nan, 3, np.nan]})
gb = df.groupby('a')

print(gb.fillna(0, inplace=True))
# Empty DataFrame
# Columns: []
# Index: []

print(gb['b'].fillna(0, inplace=True))
# a
# 1    None
# 2    None
# Name: b, dtype: object

print(df)
#    a    b
# 0  1  NaN
# 1  1  3.0
# 2  2  NaN
```

### LINKED ISSUE #55718: DEPR: groupby.fillna
Ref: https://github.com/pandas-dev/pandas/pull/53496#discussion_r1213835538

With the `method` keyword already deprecated, `groupby.fillna` provides no benefit over `Series.fillna` and `DataFrame.fillna` and is significantly less performant.

### PR DESCRIPTION
**Pull request by rhshadrach**
- [x] closes #55718
- [x] closes #48792 
- [x] closes #48395
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -973,6 +973,14 @@ def fillna(
         mouse  0.0
         dtype: float64
         """
+        warnings.warn(
```
Comment (mroeschke) [inline pandas/core/groupby/generic.py:976]: Could you add a `.. deprecated:: 2.2.0` directive near the top of the docstring?

**Review by mroeschke — COMMENTED** [review]
One comment otherwise looks good

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (rhshadrach) [reply pandas/core/groupby/generic.py:971]: Thanks - I also removed the other deprecations from the docstring. Seemed like if the entire method was deprecated we didn't need notes on individual parameters. Can restore if desired.

Comment (mroeschke) [reply pandas/core/groupby/generic.py:971]: I think that cleanup makes sense. Thanks!

File: doc/source/whatsnew/v2.2.0.rst [hunk 2]
```diff
@@ -297,6 +297,7 @@ Other Deprecations
 - Deprecated the option ``mode.data_manager`` and the ``ArrayManager``; only the ``BlockManager`` will be available in future versions (:issue:`55043`)
 - Deprecated the previous implementation of :class:`DataFrame.stack`; specify ``future_stack=True`` to adopt the future version (:issue:`53515`)
 - Deprecating downcasting the results of :meth:`DataFrame.fillna`, :meth:`Series.fillna`, :meth:`DataFrame.ffill`, :meth:`Series.ffill`, :meth:`DataFrame.bfill`, :meth:`Series.bfill` in object-dtype cases. To opt in to the future version, use ``pd.set_option("future.no_silent_downcasting", True)`` (:issue:`54261`)
+- Deprecated :meth:`.DataFrameGroupBy.fillna` and :meth:`.SeriesGroupBy.fillna`; use :meth:`.DataFrameGroupBy.ffill`, :meth:`.DataFrameGroupBy.bfill` for forward and backward filling or :meth:`.DataFrame.fillna` to fill with a single value (or the Series equivalents) :issue:`55718`
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.2.0.rst:300]:
```suggestion
- Deprecated :meth:`.DataFrameGroupBy.fillna` and :meth:`.SeriesGroupBy.fillna`; use :meth:`.DataFrameGroupBy.ffill`, :meth:`.DataFrameGroupBy.bfill` for forward and backward filling or :meth:`.DataFrame.fillna` to fill with a single value (or the Series equivalents) (:issue:`55718`)
```

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @rhshadrach

**Pull request merged**